opengemini/query_test.go

```diff
@@ -69,6 +70,70 @@ func TestQueryWithEpoch(t *testing.T) {
 		assert.Equal(t, length, getTimestampLength(v))
 	}
 }
+func TestQueryWithMsgPack(t *testing.T) {
+	c := testNewClient(t, &Config{
+		Addresses: []addr.Address{{
+			Host: "localhost",
+			Port: 8086,
+		}},
+		Encoding: MSGPACK,
+	})
+
+	// create a test database with rand suffix
+	database := randomDatabaseName()
+	err := c.CreateDatabase(database)
+	assert.Nil(t, err)
+
+	// delete test database before exit test case
+	defer func() {
+		err := c.DropDatabase(database)
+		assert.Nil(t, err)
+	}()
+
+	testMeasurement := randomMeasurement()
+	p := &Point{}
+	p.Measurement = testMeasurement
+	p.AddField("TestField", 123)
+	p.Time = time.Now()
+
+	err = c.WritePoint(database, p, func(err error) {
+		assert.Nil(t, err)
+	})
+	assert.Nil(t, err)
+
+	time.Sleep(time.Second * 5)
+
+	PrecisionTimestampLength := make(map[Precision]int64)
+	PrecisionTimestampLength[PrecisionNanosecond] = 19
+	PrecisionTimestampLength[PrecisionMicrosecond] = 16
+	PrecisionTimestampLength[PrecisionMillisecond] = 13
+	PrecisionTimestampLength[PrecisionSecond] = 10
+	PrecisionTimestampLength[PrecisionMinute] = 8
+	PrecisionTimestampLength[PrecisionHour] = 6
+
+	// check whether write success
+	for precision, length := range PrecisionTimestampLength {
+		q := Query{
+			Database:  database,
+			Command:   "select * from " + testMeasurement,
+			Precision: precision,
+		}
+		result, err := c.Query(q)
+		assert.Nil(t, err)
+		var v int64
+		switch val := result.Results[0].Series[0].Values[0][0].(type) {
```

Comment:
For json result, this type is float, please check  `v := int64(result.Results[0].Series[0].Values[0][0].(float64))`  inside the test case "TestQueryWithEpoch" . @vicky-run  @xuthus5

Reply:
Can you side confirm ?